File: Week1/simple/删除排序数组中的重复项/Solution.js
/*
执行用时：88 ms, 在所有 JavaScript 提交中击败了96.16%的用户
内存消耗：39.9 MB, 在所有 JavaScript 提交中击败了43.32%的用户
* */

/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    let i = 0, j = 1;
    while (j < nums.length){
        if (nums[i] !== nums[j]) nums[++i] = nums[j];
        j++;
    }
    return ++i;
};

console.log(removeDuplicates([1,1,2]))
console.log(removeDuplicates([0,0,1,1,2,2,3,4,4,5]))

File: Week1/simple/删除排序数组中的重复项/Solution.java
package 删除排序数组中的重复项;
/*
执行用时：0 ms, 在所有 Java 提交中击败了100.00%的用户
内存消耗：40.5 MB, 在所有 Java 提交中击败了62.35%的用户
* */
class Solution {
    public static void main(String[] args) {
//        int[] nums = {0,0,1};
        int[] nums = {0,0,1,1,1,2,2,3,3,4};
        System.out.println(removeDuplicates(nums));
    }
    public static int removeDuplicates(int[] nums) {
        int i = 0, j = 1;
        while (j < nums.length){
            if (nums[i] != nums[j]) nums[++i] = nums[j];
            j++;
        }
        return ++i;
    }
}